File: PluTodo_flutter_client/plutodo_client/lib/models/authentication/register_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'register_dto.g.dart';

@JsonSerializable()
class RegisterDto {
  RegisterDto(this.username, this.emailAddress, this.password, this.passwordConfirm);

  final String username;
  final String emailAddress;
  final String password;
  final String passwordConfirm;

  factory RegisterDto.fromJson(Map<String, dynamic> json) => _$RegisterDtoFromJson(json);
  Map<String, dynamic> toJson() => _$RegisterDtoToJson(this);
}

File: PluTodo_API_Server/Plu-Todo_API_Server/Plu-Todo_API_Server/Entities/DTOs/Auth/RegisterDTO.cs
using System.ComponentModel.DataAnnotations;

namespace Plu_Todo_API_Server.Entities.DTOs;

public class RegisterDTO
{
    [Required]
    public string Username { get; set; }

    [Required]
    [EmailAddress]
    public string Email { get; set; }

    [Required]
    public string Password { get; set; }

    [Required]
    public string PasswordConfirm { get; set; }
}